### PR TITLE
[#7319] Make `msiSetResource` not crash agent

### PR DESCRIPTION
**Pull request by MartinFlores751**
This provides a simple fix to `msiSetResource(...)` by adding a null check before dereference.

### CODE REVIEW

File: server/re/src/icatGeneralMS.cpp
Comment (korydraughn) [inline server/re/src/icatGeneralMS.cpp:144]: Use the new logger.
```cpp
#include "irods/irods_logger.hpp"

namespace
{
    using log_msi = irods::experimental::log::microservice;
} // anonymous namespace

// Replace rodsLog with ...
log_msi::error("your message ...");
```

See https://docs.irods.org/4.3.1/developers/logging/ for more info on the logging library.

File: server/re/src/icatGeneralMS.cpp
Comment (korydraughn) [inline server/re/src/icatGeneralMS.cpp:145]: Leave yourself a comment about updating this error code so you don't forget.

File: server/re/src/icatGeneralMS.cpp
Comment (korydraughn) [inline server/re/src/icatGeneralMS.cpp:143]: Swap the operands for extra protection.

File: scripts/irods/test/test_native_rule_engine_plugin.py
Comment (korydraughn) [inline scripts/irods/test/test_native_rule_engine_plugin.py:855]: `__7319` -> `__issue_7319`

**Review by korydraughn — COMMENTED** [review]
Looks good.

File: server/re/src/icatGeneralMS.cpp
Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp]: Done.

File: server/re/src/icatGeneralMS.cpp
Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp:144]: What's the benefit of the anonymous namespace? Does it keep log_msi better localized?

File: server/re/src/icatGeneralMS.cpp
Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp:143]: I'm assuming this is for preventing accidental `=` instead of `==`, right?

Comment (korydraughn) [reply server/re/src/icatGeneralMS.cpp:143]: Correct.

File: server/re/src/icatGeneralMS.cpp
Comment (korydraughn) [reply server/re/src/icatGeneralMS.cpp:144]: Correct. The compiler will mark the symbols as private to the translation unit. It's not required. It's just one way to do things.

If you prefer to move the alias into the function directly, that's perfectly fine too.

File: scripts/irods/test/test_native_rule_engine_plugin.py
Comment (MartinFlores751) [reply scripts/irods/test/test_native_rule_engine_plugin.py]: Done.

File: server/re/src/icatGeneralMS.cpp
Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp]: I was mostly just curious, since I don't think I've seen it used for this before.  Done.

File: server/re/src/icatGeneralMS.cpp
Comment (korydraughn) [inline server/re/src/icatGeneralMS.cpp:150]: `DOI` -> `DOI (DataObjInfo)`

Comment (korydraughn) [inline server/re/src/icatGeneralMS.cpp:150]: `"Cannot ..."` -> `"{}: Cannot ...", __func__`

Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp:150]: Should we just make this in plain english, "Data Object Info"?

Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp]: Done.

Comment (korydraughn) [reply server/re/src/icatGeneralMS.cpp]: No. `DataObjInfo` is a real thing in iRODS and using plain english to try and refer to it would only lead to more confusion.

Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp]: Done.

File: server/re/src/icatGeneralMS.cpp
Comment (korydraughn) [inline server/re/src/icatGeneralMS.cpp:150]: `"not found."` seems a little misleading.

How do you feel about `"Cannot set resource, no DOI (DataObjInfo) attached to REI."`?

That captures the situation a bit better IMO. Thoughts?

Comment (korydraughn) [reply server/re/src/icatGeneralMS.cpp:150]: Or is that too specific?

Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp:150]: I'm fine with the more verbose description. Should we 'full name' the REI like we did with DOI?

Comment (korydraughn) [reply server/re/src/icatGeneralMS.cpp:150]: Sure. It wouldn't hurt to put it in parens.

Comment (MartinFlores751) [reply server/re/src/icatGeneralMS.cpp]: Done.

**Comment by korydraughn** [comment]
Seems good to me.

Let's run clang-format.

Does it compile? Do the unit/core tests pass?

**Comment by korydraughn** [comment]
Thinking about this more, I don't think we need to run the unit/core tests.

**Comment by MartinFlores751** [comment]
It does compile. At the very least, I did run the new test and it passed.

Can run all tests if desired.

**Review by alanking — COMMENTED** [review]
I think the new test passing is good enough. Probably ready to squash

**Comment by MartinFlores751** [comment]
Squashed.

**Review by alanking — APPROVED** [review]
Seems good to me. Let's wait for one more approval before #'ing.

**Review by korydraughn — APPROVED** [review]
Pound it!

**Comment by MartinFlores751** [comment]
Pounded!

**Comment by korydraughn** [comment]
Nice. Let's get that cherry-pick to 4-3-stable.

**Pull request merged**
